src/app/reporting/Engine.cpp

```diff
@@ -405,7 +412,10 @@ CHIP_ERROR Engine::BuildAndSendSingleReportData(ReadHandler * apReadHandler)
         err = report.Init(reader);
         SuccessOrExit(err);
 
-        err = report.CheckSchemaValidity();
+        if ((err = report.CheckSchemaValidity()) != CHIP_NO_ERROR)
```

Comment:
add #if CHIP_CONFIG_IM_ENABLE_SCHEMA_CHECK before and after

Reply:
This is already wrapped in that macro, between line 446 and 464